domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt

```diff
@@ -1,18 +1,343 @@
+        )
+        .build()
+
+private fun findHintForProblemWithParent(path: Path): Assertion? {
+    val absolutePath = path.toAbsolutePath()
+    var currentParentPart = absolutePath.root
+    for (part in absolutePath) {
+        currentParentPart = currentParentPart.resolve(part)
+        if (currentParentPart != path) {
+            try {
+                val attributes = currentParentPart.readAttributes<BasicFileAttributes>()
+                if (!attributes.isDirectory) {
+                    return hintForParentFailure(
+                        currentParentPart,
+                        explanation = hintForNotDirectory(attributes.fileType)
+                    )
+                }
+            } catch (e: AccessDeniedException) {
+                return hintForParentFailure(
+                    currentParentPart.parent,
+                    explanation = hintForAccessDenied(currentParentPart.parent)
+                )
+            } catch (e: IOException) {
+                return hintForParentFailure(
+                    currentParentPart,
+                    explanation = hintForFileSpecificIoException(currentParentPart, e)
+                )
+            }
+        }
+    }
+    return null
+}
+
+private fun hintForNoSuchFile() =
+    ExpectImpl.builder.explanatory
+        .withExplanation(FAILURE_DUE_TO_NO_SUCH_FILE)
+        .build()
+
+private fun hintForNotDirectory(actualType: Translatable) =
+    ExpectImpl.builder.explanatory
+        .withExplanation(FAILURE_DUE_TO_WRONG_FILE_TYPE, actualType, A_DIRECTORY)
+        .build()
+
+private fun hintForParentFailure(parent: Path, explanation: Assertion) =
+    ExpectImpl.builder.explanatoryGroup.withDefaultType
+        .withAssertions(
+            ExpectImpl.builder.descriptive.failing
+                .withDescriptionAndRepresentation(FAILURE_DUE_TO_PARENT, parent)
+                .build(),
+            when (explanation) {
+                is AssertionGroup -> explanation
+                else -> ExpectImpl.builder.explanatoryGroup.withDefaultType
+                    .withAssertion(explanation)
+                    .build()
+            }
+        ).build()
+
+private fun hintForOtherIoException(exception: IOException) =
+    ExpectImpl.builder.explanatory
+        .withExplanation(
+            FAILURE_DUE_TO_EXCEPTION,
+            exception::class.simpleName ?: "IoException",
+            exception.message ?: ""
+        )
+        .build()
+
+private val Path.fileType: Translatable
+    get() = readAttributes<BasicFileAttributes>().fileType
+
+private val BasicFileAttributes.fileType: Translatable
+    get() = when {
+        isRegularFile -> A_FILE
+        isDirectory -> A_DIRECTORY
+        isSymbolicLink -> A_SYMBOLIC_LINK
+        else -> A_UNKNOWN_FILE_TYPE
+    }
+
+private fun formatPosixPermissions(filePermissions: Set<PosixFilePermission>): StringBuilder {
+    val permissionString = StringBuilder(3 * 5 + 2)
+    permissionString
```

Comment:
This _is_ the order `ls -la` uses: owner, group and others. You might be irritated because of the different wording in Java and Linux:

Linux | Java
---|---
user | owner
group | group
others | others

The prefixes are the ones `chmod` accepts:

> A  combination of the letters `ugoa` controls which users' access to the file will be changed: the user who owns it (`u`), other users in the file's group (`g`), other users not in the  file's  group (`o`), or all users (`a`).

source: `man chmod`

Reply:
Apparently a misunderstanding on my part, thanks for the explanation 👍